cartographer_ros/cartographer_ros/node_main.cc

```diff
@@ -38,14 +41,20 @@ void Run() {
   tf2_ros::TransformListener tf(tf_buffer);
   NodeOptions node_options;
   TrajectoryOptions trajectory_options;
-  std::tie(node_options, trajectory_options) =
-      LoadOptions(FLAGS_configuration_directory, FLAGS_configuration_basename);
-
+  if (FLAGS_start_trajectory_with_default_topics) {
+    std::tie(node_options, trajectory_options) = LoadOptions(
+        FLAGS_configuration_directory, FLAGS_configuration_basename);
+  } else {
+    node_options = LoadNodeOptions(FLAGS_configuration_directory,
```

Comment:
I see, that makes sense.

In that case, it seems to me that LoadOptions and LoadNodeOptions should be moved into node_main.cc to remove the duplication of loading the LuaParameterDictionary in LoadOptions and LoadNodeOptions. It would also remove the confusing naming of those methods.

You would simply load the Lua configuration, then call CreateNodeOptions and CreateTrajectoryOptions as appropriate all directly in node_main.cc.